Source/Core/Logging/FramesEventLogger.swift

```diff
@@ -14,24 +14,46 @@ protocol FramesEventLogging {
 
 final class FramesEventLogger: FramesEventLogging {
 
-    private let correlationID: String
+    private let getCorrelationID: () -> String
     private let checkoutEventLogger: CheckoutEventLogging
     private let dateProvider: DateProviding
     
     // MARK: - Init
+
+    convenience init(environment: Environment, getCorrelationID: @escaping () -> String) {
+        let checkoutEventLogger = CheckoutEventLogger(productName: Constants.productName)
+        let appBundle = Foundation.Bundle.main
+        let appPackageName = appBundle.bundleIdentifier ?? "unavailableAppPackageName"
+        let appPackageVersion = appBundle.infoDictionary?["CFBundleShortVersionString"] as? String ?? "unavailableAppPackageVersion"
+
+        let uiDevice = UIKit.UIDevice.current
```

Comment:
Unnecessary source framework mentioned, as well as another with `Foundation` for `appBundle`. I'm curious how these ones come to be as I don't think autocomplete normally adds them (although recently it does the auto import)